pkg/cmd/roachprod/cli/chaos.go

```diff
@@ -0,0 +1,395 @@
+		Use:   "chaos [command]",
+		Short: "Failure injection related commands",
+		Long: `Failure injection related commands for testing cluster resilience.
+
+Failure injection commands allow you to inject various types of failures into a
+cluster to test its behavior under adverse conditions. Each failure type has its
+own lifecycle: Setup → Inject → Wait → Recover → Cleanup.
+
+Global flags control the duration and cleanup behavior of all chaos commands.
+`,
+	}
+
+	// Add global flags
+	chaosCmd.PersistentFlags().DurationVar(&chaosWaitBeforeCleanup,
+		"wait-before-cleanup", 5*time.Minute,
+		"time to wait before cleaning up the failure")
+	chaosCmd.PersistentFlags().BoolVar(&chaosRunForever,
+		"run-forever", false,
+		"if set, takes precedence over --wait-before-cleanup. On graceful shutdown, cleans up the injected failure")
+	chaosCmd.PersistentFlags().StringVar(&chaosCertsDir,
+		"certs-dir", install.CockroachNodeCertsDir,
+		"local path to certs directory for secure clusters")
+	chaosCmd.PersistentFlags().IntVar(&chaosReplicationFactor,
+		"replication-factor", 0,
+		"expected replication factor for the cluster (0 = use default of 3)")
+	chaosCmd.PersistentFlags().StringVar(&chaosStage,
+		"stage", string(StageAll),
+		`lifecycle stage to execute. Options:
+  - all: runs the complete lifecycle (Setup → Inject → Wait → Recover → Cleanup)
+  - setup: runs only the setup phase (prepares failure dependencies)
+  - inject: runs only the inject phase (activates the failure)
+  - recover: runs only the recover phase (removes the failure)
+  - cleanup: runs only the cleanup phase (removes failure dependencies)
+Default: all`)
+
+	// Add subcommands
+	chaosCmd.AddCommand(cr.buildChaosNetworkCmd())
+
+	return chaosCmd
+}
+
+// getGlobalChaosOpts returns the global chaos options from flags.
+// It validates the stage flag and returns an error if invalid.
+func getGlobalChaosOpts() (GlobalChaosOpts, error) {
+	stage := FailureStage(chaosStage)
+
+	// Validate stage
+	validStages := ValidStages()
+	isValid := false
+	for _, s := range validStages {
+		if string(stage) == s {
+			isValid = true
+			break
+		}
+	}
+	if !isValid {
+		return GlobalChaosOpts{}, errors.Newf("invalid stage %q, must be one of: %s",
+			chaosStage, strings.Join(validStages, ", "))
+	}
+
+	return GlobalChaosOpts{
+		WaitBeforeCleanup: chaosWaitBeforeCleanup,
+		RunForever:        chaosRunForever,
+		Stage:             stage,
+	}, nil
+}
+
+// getClusterOptions returns cluster options for creating a failer
+func getClusterOptions() []failures.ClusterOptionFunc {
+	opts := []failures.ClusterOptionFunc{
+		failures.Secure(!insecure),
+		failures.LocalCertsPath(chaosCertsDir),
+	}
+	if chaosReplicationFactor > 0 {
+		opts = append(opts, failures.ReplicationFactor(chaosReplicationFactor))
+	}
+	return opts
+}
+
+// parseInt32SliceToNodes converts a uint32 slice to install.Nodes
+func parseInt32SliceToNodes(nodes []int32) install.Nodes {
+	if len(nodes) == 0 {
```

Comment:
nit: this check is unnecessary since the for loop should handle the zero length case